### ISSUE TITLE
Random RuntimeException: Another cursor derived from this client is in exhaust

### ISSUE THREAD

**Issue by RobIsHere** [mongodb/mongo-php-library#178]
```
        $cursor = Store::DB()->collection($this->GetCollection())->find($query);
        $list = [];
        foreach ($cursor as $d)
        {
            $list[] = $d;
        }
```

MongoDB\Driver\Exception\RuntimeException: Another cursor derived from this client is in exhaust

By the way: this error message is simply a bug itself. English words. But very strange message points me towards what? ;)

Server is up and running. Other queries succeed.

**Comment by RobIsHere** [comment]
I experimented with removing the $query parameter in the code above. Now the apache log says:

child pid 64953 exit signal Segmentation fault (11)

**Comment by RobIsHere** [comment]
If i do inside of Store::DB()->collection():
```
$client = new \MongoDB\Client(null, [], ['typeMap' => ['root' => 'array', 'document' => 'array', 'array' => 'array']]);
$cursor = $client->selectCollection(Store::DB()->name, $collectionName);
return $cursor;
```
The $cursor->find() segfaults or has this weired error.

If i put the same ($collectionName becomes ->GetCollection():collectionName) code above the find() call, it works.

**Comment by jmikola** [comment]
Can you share the MongoDB and driver versions in use? `phpinfo()` should report version information for the driver and the libmongoc and libbson libraries it is using.

> MongoDB\Driver\Exception\RuntimeException: Another cursor derived from this client is in exhaust
>
> By the way: this error message is simply a bug itself. English words. But very strange message points me towards what? ;)

That error message emanates from [libmongoc](https://github.com/mongodb/mongo-c-driver/blob/1.3.5/src/mongoc/mongoc-cursor.c#L980), but I would only expect it if another cursor on the same client instance was opened with the exhaust option (discussed [here](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#op-query)).

> I experimented with removing the $query parameter in the code above. Now the apache log says:
> 
> child pid 64953 exit signal Segmentation fault (11)

Are you able to get a backtrace of this segfault? That may be a bug in the extension.

I'll note that since you opened this issue, we released [1.1.7](https://pecl.php.net/package/mongodb/1.1.7) of the extension, which fixed at least one segfault related to cursor usage.

----

Unrelated to the actual problem, I would suggest using [MongoDB\Driver\Cursor::toArray()](http://php.net/manual/en/mongodb-driver-cursor.toarray.php) or simply `iterator_to_array()` in lieu of `foreach ($cursor as $d) { $list[] = $d; }` if you'd like to convert the cursor into an array of documents.

**Comment by pietercolpaert** [comment]
Experienced the same issue with the 1.1.5 version of the driver. When updating to 1.1.8, everything worked fine!

**Comment by jmikola** [comment]
@RobIsHere: Closing this due to inactivity and in light of @pietercolpaert's comment that a similar error may have been resolved in 1.1.8. If the issue persists and you're able to address my last questions, feel free to re-open.

Should the issue persist, I'd appreciate if you could attempt reproduction on driver version [1.2.0alpha2](https://github.com/mongodb/mongo-php-driver/releases/tag/1.2.0alpha2). This was recently released and includes significant changes to our connection handling and persistence along with a bump to our dependency on libmongoc. Thanks!
